src/OmniSharp.Abstractions/Services/IAssemblyLoader.cs

```diff
@@ -54,11 +55,22 @@ public static IEnumerable<Assembly> Load(this IAssemblyLoader loader, params str
             }
         }
 
-        public static IEnumerable<Assembly> LoadByAssemblyNameOrPath(this IAssemblyLoader loader, IEnumerable<string>  assemblyNames)
+        public static IEnumerable<Assembly> LoadByAssemblyNameOrPath(this IAssemblyLoader loader, ILogger logger, IEnumerable<string>  assemblyNames)
```

Comment:
Can we inject a `LoggerFactory` into `AssemvlyLoader` and create logger and log there instead? Feels wrong to pass the logger around.

Reply:
@mholo65 To clarify, are you proposing to pass ILoggerFactory instead of ILogger into LoadByAssemblyNameOrPath?

Reply:
@mholo65 Switched to passing in ILoggerFactory - please take another look. Thanks!

Reply:
No, what I meant was injecting `ILoggerFactory` to `AssemblyLoader`. But it seems like we are already doing this (see [AssemblyLoader.cs](https://github.com/OmniSharp/omnisharp-roslyn/blob/a3e7dbf7e8b6cc14e0c43879c53d940e5910a892/src/OmniSharp.Host/Services/AssemblyLoader.cs)). Why do we need additional logging in the extension method, isn't the logging in the implementation enough? It looks like we already catching and logging all exceptions. So this `catch` will never hit.

Reply:
@mholo65 Below is the callstack where the exception gets thrown if plugin assembly cannot be loaded. With the way the code is structured regardless of my change, the plugins loading code path is going through extension methods, not throw the actual AssemblyLoader implementation. Because added exception catching and logging attempts to address the existing problem with plugins loading I can leave it out of this change. Please advise.

System.IO.FileLoadException
  HResult=0x80131047
  Message=The given assembly name or codebase was invalid. (Exception from HRESULT: 0x80131047)
  Source=mscorlib
  StackTrace:
   at System.Reflection.AssemblyName.nInit(RuntimeAssembly& assembly, Boolean forIntrospection, Boolean raiseResolveEvent)
   at System.Reflection.AssemblyName..ctor(String assemblyName)
   at OmniSharp.Services.IAssemblyLoaderExtensions.Load(IAssemblyLoader loader, String name) in C:\src\omnisharp-roslyn\src\OmniSharp.Abstractions\Services\IAssemblyLoader.cs:line 33

Reply:
Ah, so this is where it actually throws https://github.com/OmniSharp/omnisharp-roslyn/blob/6d3ce87c502264453507f1669314c412a44acc1b/src/OmniSharp.Abstractions/Services/IAssemblyLoader.cs#L33

The provided path is neither a file that exists nor a valid assembly name. Sorry, don't have a quick solution for providing valuable information about the assembly that it failed to load. Maybe passing the `ILogger` is the best option for now.

Reply:
@mholo65 Thanks - I reverted the last commit to go back to passing in ILogger. Please let me know if you have any other concerns.